File: Algorithms/Searching-Algorithm/Jump-Search/jump_search.py
import math 
  
def jumpSearch( arr , x , n ): 
      
    step = math.sqrt(n) 
    
    prev = 0
    while arr[int(min(step, n)-1)] < x: 
        prev = step 
        step += math.sqrt(n) 
        if prev >= n: 
            return -1
      
     
    while arr[int(prev)] < x: 
        prev += 1
          
     
        if prev == min(step, n): 
            return -1
      
    if arr[int(prev)] == x: 
        return prev 
      
    return -1

arr = [ 0,0, 1, 1, 2, 4, 5, 8, 17, 21,29, 
    34, 70, 89, 200, 377] 
x = 200
n = len(arr) 
  
index = jumpSearch(arr, x, n) 

if(index == -1):
    print("Number" , x , "is not present in array")
else:
    print("Number" , x, "is at index" ,"%.0f"%index) 

File: Algorithms/Searching-Algorithm/Jump-Search/jumpSearch.go
//Jump Search implementation in Golang.

package main
import "fmt"
import "math"


func main() {

  arr := []int{ 0, 1, 2, 4, 9, 16, 25, 36, 64, 81, 100, 121, 144, 189, 400, 625}
  n:= len(arr)
  x:= 144


  z:= jumpSearch(arr, x, n)
  fmt.Println("Number found at index",z)

}

func jumpSearch(arr []int, x int, n int) (inx float64){

  step:= math.Round(math.Sqrt(float64(n)))
  var i float64
  for i= 0; i < float64(n); i+=step {
    if (arr[int(i)] > x){
      break;
    }
  }
  i = i - step
  ub:= i + step
  for ; i < ub; i++{
    if(arr[int(i)]==x){
      return i;
    }
  }
  return -1;
}
